relayer/workers/ethrelayer/main.go

```diff
@@ -0,0 +1,165 @@
+package ethrelayer
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/sirupsen/logrus"
+	"github.com/snowfork/go-substrate-rpc-client/v2/types"
+	"github.com/snowfork/polkadot-ethereum/relayer/chain"
+	"github.com/snowfork/polkadot-ethereum/relayer/chain/ethereum"
+	"github.com/snowfork/polkadot-ethereum/relayer/chain/parachain"
+	"github.com/snowfork/polkadot-ethereum/relayer/crypto/sr25519"
+)
+
+type Worker struct {
+	parachainConfig  *parachain.Config
+	ethereumConfig   *ethereum.Config
+	parachainConn    *parachain.Connection
+	ethereumConn     *ethereum.Connection
+	ethereumListener *EthereumListener
+	parachainWriter  *ParachainWriter
+	log              *logrus.Entry
+}
+
+const Name = "eth-relayer"
+
+func NewWorker(ethereumConfig *ethereum.Config, parachainConfig *parachain.Config) (*Worker, error) {
+	log := logrus.WithField("worker", Name)
+
+	log.Info("Creating worker")
+
+	ethereumConn := ethereum.NewConnection(ethereumConfig.Endpoint, nil, log)
+
+	// Generate keypair from secret
+	parachainKeypair, err := sr25519.NewKeypairFromSeed(parachainConfig.PrivateKey, "")
+	if err != nil {
+		return nil, err
+	}
+	parachainConn := parachain.NewConnection(parachainConfig.Endpoint, parachainKeypair.AsKeyringPair(), log)
+
+	// channel for messages from ethereum
+	ethMessages := make(chan []chain.Message, 1)
+	// channel for headers from ethereum (it's a blocking channel so that we
+	// can guarantee that a header is forwarded before we send dependent messages)
+	ethHeaders := make(chan chain.Header)
+
+	ethereumListener, err := NewEthereumListener(ethereumConfig, ethereumConn, ethMessages, ethHeaders, log)
+	if err != nil {
+		return nil, err
+	}
+
+	parachainWriter, err := NewParachainWriter(parachainConn, ethMessages, ethHeaders, log)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Worker{
+		ethereumConfig:   ethereumConfig,
+		ethereumConn:     ethereumConn,
+		parachainConfig:  parachainConfig,
+		parachainConn:    parachainConn,
+		ethereumListener: ethereumListener,
+		parachainWriter:  parachainWriter,
+		log:              log,
+	}, nil
+
+}
+
+func (worker *Worker) Start(ctx context.Context, eg *errgroup.Group) error {
+	worker.log.Info("Starting worker")
+
+	if worker.ethereumListener == nil || worker.parachainWriter == nil {
+		return fmt.Errorf("Sender and/or receiver need to be set before starting chain")
+	}
+
+	err := worker.ethereumConn.Connect(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Short-lived channels that communicate initialization parameters
+	// between the two chains. The chains close them after startup.
+	subInit := make(chan chain.Init)
```

Comment:
This seems unused. Was it ever used or is it safe to delete? @Rizziepit @denalimarsh

Reply:
Yip get rid of it. It was only relevant in the `Chain` world where one chain doesn't have access to the connection of another chain. I got rid of `ethInit` in https://github.com/Snowfork/polkadot-ethereum/pull/354/ too